src/util/permissions.ts

```diff
@@ -1,256 +1,286 @@
   }) as Permission[];
 }
 
 /** This function converts an array of permissions into the bitwise string. */
 export function calculateBits(permissions: Permission[]) {
   return permissions.reduce(
+    // Get the bit value for this permission and assign it to bits
     (bits, perm) => bits |= BigInt(Permissions[perm]),
     BigInt(0),
   ).toString();
 }
 
+/** Gets the highest role from the member in this guild */
 export async function highestRole(guildID: string, memberID: string) {
   const guild = await cacheHandlers.get("guilds", guildID);
-  if (!guild) return;
-
-  const member = (await cacheHandlers.get("members", memberID))?.guilds.get(
-    guildID,
-  );
-  if (!member) return;
+  if (!guild) throw Error(Errors.GUILD_NOT_FOUND);
+  // Get the roles from the member
+  const memberRoles = (await cacheHandlers.get("members", memberID))?.guilds
+    .get(
+      guildID,
+    )?.roles;
+  // This member has no roles so the highest one is the @everyone role
+  if (!memberRoles) return guild.roles.get(guildID) as Role;
 
   let memberHighestRole: Role | undefined;
 
-  for (const roleID of member.roles) {
+  for (const roleID of memberRoles) {
     const role = guild.roles.get(roleID);
+    // Rare edge case handling if undefined
     if (!role) continue;
-
+    // If memberHighestRole is still undefined we want to assign the role,
+    // else we want to check if the current role position is higher than the current memberHighestRole
     if (
       !memberHighestRole || memberHighestRole.position < role.position
     ) {
       memberHighestRole = role;
     }
   }
 
-  return memberHighestRole || (guild.roles.get(guild.id) as Role);
+  // The member has at least one role so memberHighestRole must exist
+  return memberHighestRole!;
 }
 
+/** Checks if the first role is higher than the second role */
 export async function higherRolePosition(
   guildID: string,
   roleID: string,
   otherRoleID: string,
 ) {
   const guild = await cacheHandlers.get("guilds", guildID);
-  if (!guild) return;
+  if (!guild) throw Error(Errors.GUILD_NOT_FOUND);
 
   const role = guild.roles.get(roleID);
   const otherRole = guild.roles.get(otherRoleID);
-  if (!role || !otherRole) return;
-
+  if (!role || !otherRole) throw Error(Errors.ROLE_NOT_FOUND);
   // Rare edge case handling
   if (role.position === otherRole.position) {
     return role.id < otherRole.id;
   }
 
   return role.position > otherRole.position;
 }
+
+/** Checks if the member has a higher position than the given role */
+export async function isHigherPosition(
+  guildID: string,
+  memberID: string,
+  compareRoleID: string,
+) {
+  const guild = await cacheHandlers.get("guilds", guildID);
+  if (!guild) throw Error(Errors.GUILD_NOT_FOUND);
+
+  if (guild.ownerID === memberID) return true;
+
+  const memberHighestRole = await highestRole(guildID, memberID);
+  const compareRole = guild.roles.get(compareRoleID);
```

Comment:
Can't you just `return higherRolePosition(guildID, memberHighestRole.id, compareRoleID)`